fe/fe-core/src/main/java/com/starrocks/qe/ShowExecutor.java

```diff
@@ -1952,6 +1955,46 @@ private void handleShowRestore() throws AnalysisException {
         resultSet = new ShowResultSet(showStmt.getMetaData(), infos);
     }
 
+    private String getCatalogNameById(long catalogId) {
+        if (CatalogMgr.isInternalCatalog(catalogId)) {
+            return InternalCatalog.DEFAULT_INTERNAL_CATALOG_NAME;
+        }
+
+        CatalogMgr catalogMgr = GlobalStateMgr.getCurrentState().getCatalogMgr();
+        Optional<Catalog> catalogOptional = catalogMgr.getCatalogById(catalogId);
+        if (!catalogOptional.isPresent()) {
+            throw new SemanticException("cannot find catalog");
```

Comment:
Use MetaNotFoundException, this can avoid unnecessary error reporting due to catalog deletion

Reply:
my mistake, will be modified in next pr